### PR TITLE
Milestone: #Capybara specs-pair programming

### PR DESCRIPTION
**Pull request by cosywasswa**
👇 In this milestone, we implemented the following.

- [ ] Fixed the N+1 problem in the controllers
- [ ] Used Capybara to write integration tests for each view.
- [ ] Grouped tests according to the requirements.
- [ ] Wrote tests for users_index, user_show, posts_index, post_show pages.
- [ ] Updated README file with project updates.
- [ ] Fixed linter bugs
Note: attached are N+1 screenshots.
posts-before

posts-after

users-after

users-before

### CODE REVIEW

File: spec/features/user_show_spec.rb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  end
+  describe 'GET/posts/show' do
+    it 'When I click a user post, it redirects me to that post show page' do
+      visit user_path(@user1)
+
+      post = @user1.recent_posts.first
+      click_link(post.id)
+      expect(page).to have_current_path(user_post_path(@user1, post))
+    end
+  end
```
Comment (DammyShittu) [inline spec/features/user_show_spec.rb]: - [ ] Kindly add test that when we click the "See all posts" button it takes the user to the posts index page.

File: spec/features/post_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      end
+    end
+  end
+  it 'I can see the comment each commentor left.' do
+    @posts.each do |post|
+      post.recent_comments.each do |comment|
+        expect(page).to have_content(comment.user.text)
+      end
+    end
+  end
```
Comment (DammyShittu) [inline spec/features/post_show_spec.rb]: - [ ] Kindly add the tast to check that you can the post body. It is missing here.

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @cosywasswa and @manzitresor,

## Highlights
✔️ Passing tests

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/features/user_show_spec.rb [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+      post = @user1.recent_posts.first
+      click_link(post.id)
+      expect(page).to have_current_path(user_post_path(@user1, post))
+    end
+
+    it 'When I click to see all posts, it redirects me to the user posts index page.' do
+      visit user_posts_path(@user1)
+      expect(page).to have_current_path(user_posts_path(@user1))
+    end
```
Comment (DammyShittu) [inline spec/features/user_show_spec.rb]: - [ ] This test does not verify what is expected. You are visiting the post index page on line 70 and on line 71 you are expecting that the path is the index page path. That is redundant, don't you think so?

  You need to **CLICK** the "See all posts" button and ensure it redirects to the user's post index page. Take note of the click and redirect requirement

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @cosywasswa and @manzitresor,

## Highlights
✔️ Passing tests

Great work on making the changes requested 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by V-Blaze — APPROVED** [review]
Hi @cosywasswa  , 

# Approved 🟢 🟢 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
giphy

## Highlights

- All tests are passing💯
-  Covers all required test cases👍
-  fixed n+1 problems🟢 👍

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**